deploy-service/teletraanservice/src/main/java/com/pinterest/teletraan/worker/DeployTagWorker.java

```diff
@@ -122,14 +124,20 @@ private void processBatch() throws Exception {
                 LOG.info("process job: {}", job);
                 String lockName = String.format("DeployTagWorker-%s", job.getConstraint_id());
                 Connection connection = utilDAO.getLock(lockName);
+                DeployConstraintBean latestJob = deployConstraintDAO.getById(job.getConstraint_id());
                 if (connection != null) {
                     try {
-                        processEachEnvironConstraint(job);
+                        processEachEnvironConstraint(latestJob);
                     } catch (Exception e) {
-                        LOG.error("failed to process job: {}", job.toString(), e);
-                        job.setState(TagSyncState.ERROR);
-                        deployConstraintDAO.updateById(job.getConstraint_id(), job);
-                        LOG.error("updated job state to {}", TagSyncState.ERROR);
+                        LOG.error("failed to process job: {} Error {} stack {}", latestJob.toString(),
+                                ExceptionUtils.getRootCauseMessage(e), ExceptionUtils.getFullStackTrace(e));
+                        if (e instanceof SQLException) {
+                            // Don't do anything
```

Comment:
when it's an SQL exception, shall we add some messages to the log?

Reply:
Line #132 should capture the error regardless. 
That said, updated code to just `if` instead of an empty `if` followed by an else